lib/inspector/inspector_service.dart

```diff
@@ -94,6 +94,54 @@ class InspectorService {
     );
   }
 
+  /// As we aren't running from an IDE, we don't know exactly what the pub root
+  /// directories are for the current project so we make a best guess if needed
+  /// based on the the root directory of the first non artifical widget in the
+  /// tree.
+  Future<String> inferPubRootDirectoryIfNeeded() async {
+    final group = createObjectGroup('temp');
+    RemoteDiagnosticsNode root = await group.getRoot(FlutterTreeType.widget);
+
+    if (root == null) {
+      // No need to do anything as there isn't a valid tree (yet?).
+      await group.dispose();
+      return null;
+    }
+    var children = await root.children;
+    if (children?.isNotEmpty == true) {
+      return null;
+    }
+    final List<RemoteDiagnosticsNode> allChildren =
+        await group.getChildren(root.dartDiagnosticRef, false, null);
+    var path = allChildren.first.creationLocation?.path;
+    if (path == null) {
+      await group.dispose();
+      return null;
+    }
+    // TODO(jacobr): use the list of loaded scripts to determine the appropriate
```

Comment:
I think `Isolate.rootLib` is what you want - I don't think you'll need the full list of scripts (it's also available now however, as part of the new `getScripts()` call).

Reply:
I've updated the todo to use Isolate.rootLib. TLDR; I need to cleanup the --track-widget-creation transformer to stop using absolute paths.
@terrylucas  maybe you are interested in helping fix that in the transformer as you need to tweak the transformer code for some memory tracking items.